x-pack/plugins/security_solution/public/common/components/super_date_picker/selectors.ts

```diff
@@ -8,19 +8,30 @@
 import { isEmpty } from 'lodash';
 import { createSelector } from 'reselect';
 import type { State } from '../../store';
-import type { InputsModelId } from '../../store/inputs/constants';
-import type { Policy, InputsRange, TimeRange, GlobalQuery } from '../../store/inputs/model';
+import type {
+  GlobalQuery,
+  GlobalKqlQuery,
+  InputsRange,
+  InputsRangeTimeOnly,
+  Policy,
+  TimeRange,
+} from '../../store/inputs/model';
 
-export const getPolicy = (inputState: InputsRange): Policy => inputState.policy;
+export const getPolicy = (inputState: InputsRange | InputsRangeTimeOnly): Policy =>
+  inputState.policy;
 
-export const getTimerange = (inputState: InputsRange): TimeRange => inputState.timerange;
+export const getTimerange = (inputState: InputsRange | InputsRangeTimeOnly): TimeRange =>
+  inputState.timerange;
 
 export const getQueries = (inputState: InputsRange): GlobalQuery[] => inputState.queries;
 
-export const getGlobalQueries = (state: State, id: InputsModelId): GlobalQuery[] => {
+export const getGlobalQueries = (state: State, id: 'global' | 'timeline'): GlobalQuery[] => {
   const inputsRange = state.inputs[id];
   return !isEmpty(inputsRange.linkTo)
     ? inputsRange.linkTo.reduce<GlobalQuery[]>((acc, linkToId) => {
+        if (linkToId === 'socTrends') {
```

Comment:
Might be worth a larger team convo, but I wonder if it's worth it splitting up `linkTo` for the timerange and query everywhere? So a user can choose to only link the timerange for timeline and global state for instance, but still keep separate queries? Only thing there would be worrying about the url getting slightly out of hand

Reply:
so creating a new redux parent state instead of `input`? have `timerange` and `query` at top level? idk that i agree with that, but definitely not in this PR, trying to keep this reasonably sized... lets bring it up at engineering sync and i can do a follow up pr if thats what we decide

Reply:
I made a note for discussion on the 9/7 sync